benchmark/json_schema/bench_sglang.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import json
+import time
+
+from datasets import load_dataset
+
+import sglang as sgl
+from sglang.lang.chat_template import get_chat_template_by_model_path
+from sglang.test.test_utils import (
+    add_common_sglang_args_and_parse,
+    select_sglang_backend,
+)
+from sglang.utils import dump_state_text
+
+
+@sgl.function
+def schema_gen(s, message: str, json_schema: str):
+    s += message
+    s += sgl.gen("json_output", max_tokens=256, json_schema=json_schema)
+
+
+def convert_dataset(path: str):
+    tmpl = get_chat_template_by_model_path("Llama-3.1-8B-Instruct")
+
+    raw_dataset = load_dataset(path)
+    dataset = []
+    for data in raw_dataset["train"]:
+        messages = data["prompt"]
+        schema = data["schema"]
+        obj = json.loads(schema)
+        if obj.get("type", None) is None:
+            continue
+        message = tmpl.get_prompt(messages)
+
+        dataset.append(
+            {
+                "message": message,
+                "json_schema": schema,
+            }
+        )
+
+    return dataset
+
+
+def bench_schema(args):
+    arguments = convert_dataset(args.data_path)
+
+    if args.num_jsons == -1:
+        args.num_jsons = len(arguments)
+    arguments = arguments[: args.num_jsons]
+
+    # Select backend
+    backend = select_sglang_backend(args)
+    sgl.set_default_backend(backend)
+
+    # Run requests
+    tic = time.time()
+    states = schema_gen.run_batch(
+        arguments,
+        temperature=0,
+        num_threads=args.parallel,
+        progress_bar=True,
+    )
```

Comment:
need to assert the output is a correct json and has the required keys.
https://github.com/sgl-project/sglang/blob/b275ce00439b07c8d73d9f11b6d0fb74dda46629/test/srt/test_json_constrained.py#L71-L78